akka-persistence-typed/src/main/scala/akka/persistence/typed/Retention.scala

```diff
@@ -32,7 +32,7 @@ final case class RetentionCriteria(snapshotEveryNEvents: Long, keepNSnapshots: L
    */
   def toSequenceNumber(lastSequenceNr: Long): Long = {
     // Delete old events, retain the latest
-    lastSequenceNr - (keepNSnapshots * snapshotEveryNEvents)
+    math.max(0, lastSequenceNr - (keepNSnapshots * snapshotEveryNEvents))
```

Comment:
This is curious because `math.max` is in the code this uses
https://github.com/akka/akka/blob/master/akka-persistence-typed/src/main/scala/akka/persistence/typed/internal/ExternalInteractions.scala#L159

So this change would mean
```scala
protected def internalDeleteSnapshots(toSequenceNr: Long): Unit = {
    val deleteTo = toSequenceNr - 1
    val func = math.max(0, toSequenceNr - (keepNSnapshots * snapshotEveryNEvents))
    val deleteFrom = math.max(0, func)
    val snapshotCriteria = SnapshotSelectionCriteria(minSequenceNr = deleteFrom, maxSequenceNr = deleteTo)
  ....
}
```

From what I think is currently
```scala
protected def internalDeleteSnapshots(toSequenceNr: Long): Unit = {
    val deleteTo = toSequenceNr - 1
    val func = toSequenceNr - (keepNSnapshots * snapshotEveryNEvents)
    val deleteFrom = math.max(0, func)
    val snapshotCriteria = SnapshotSelectionCriteria(minSequenceNr = deleteFrom, maxSequenceNr = deleteTo)
  ....
}
```

Reply:
Maybe remove the scaladoc on this method that describes A. more logic than this method and B. the exact logic that is in this method?

Reply:
I'm not sure I understand what you mean? I have changed that `internalDeleteSnapshots` in this PR.

I placed the `max(0` here because otherwise I would have to do that on several places. Negative sequenceNr doesn't make much sense. First event has nr 1.